pkg/controller/inference/playground_controller.go

```diff
@@ -308,17 +303,40 @@ func buildWorkerTemplate(models []*coreapi.OpenModel, playground *inferenceapi.P
 	return template
 }
 
-func setPlaygroundCondition(playground *inferenceapi.Playground, service *inferenceapi.Service) {
-	// For the start up.
-	if len(playground.Status.Conditions) == 0 || !apimeta.IsStatusConditionTrue(service.Status.Conditions, inferenceapi.PlaygroundProgressing) {
+func handleUnexpectedCondition(playground *inferenceapi.Playground, modelExists bool, serviceHasOwner bool) (changed bool) {
+	// Put it in the first place as more serious.
+	if !serviceHasOwner {
```

Comment:
If the service is created by user, it shouldn't be owned by other objects. so the variable name may need to be changed.